plugin/execCommand.go

```diff
@@ -0,0 +1,128 @@
+package plugin
+
+import (
+	"context"
+	"fmt"
+)
+
+// ExecCommandImpl implements the plugin.ExecCommand interface.
+// Use plugin.NewExecCommand to create instances of these objects.
+type ExecCommandImpl struct {
+	ctx           context.Context
+	outputCh      chan ExecOutputChunk
+	stdout        *OutputStream
+	stderr        *OutputStream
+	exitCodeCh    chan int
+	exitCodeErrCh chan error
+}
+
+// NewExecCommand creates a new ExecCommandImpl object whose
+// lifetime is tied to the passed-in execution context.
```

Comment:
We should document what other functions you need to call to successfully use ExecCommandImpl.